### DISCUSSION TITLE
Microsoft Edge baseado no Chromium - O que vocês acham?

### DISCUSSION THREAD

**Discussion by fabiorocha** [frontendbr/forum#1307] [Discussão]
Disclaimer: trabalho em um dos times por trás desse esforço, mas não estou aqui como representante da empresa que trabalho. Estou interessado nas opiniões pessoais de vocês, parte da comunidade frontend brasileira.

Opa pessoal, tudo bom? Curioso pra saber o que vocês acham dessa notícia: https://blogs.windows.com/windowsexperience/2018/12/06/microsoft-edge-making-the-web-better-through-more-open-source-collaboration/. Resumindo, o Microsoft Edge vai ser mais um navegador rodando emcima da stack do [Chromium](https://www.chromium.org/), ou parte dela, como o Chrome, Opera, Vivaldi, etc.

Também me disponho a responder perguntas sobre o assunto, desde que eu possa/saiba falar sobre.  Críticas também são bem-vindas. 😊

**Comment by luizbills** [comment]
Sou totalmente a favor. Menos um navegador para testar praticamente.

**Comment by luizbills** [comment]
Sem falar que poderemos contar com a microsoft para ajudar o time de desenvolvimento do chromium para ajudar a melhorar e implementar novas features.

**Comment by fabiorocha** [comment]
@luizbills em termos de compatibilidade acho que vai ajudar bastante realmente, e é uma das principais intenções. 

Apesar de certo medo da comunidade de a plataforma estagnar por falta de concorrência, acho que, como você disse, as discussões serão agora a nível de plataforma antes de uma API ser implementada, o que acho que vai continuar movendo a plataforma. Se não acontecer, sempre dá pra fazer um fork e seguir em frente.

Não sei muito certo se falta de múltiplas implementações do padrão vai gerar um impacto negativo no ecossistema -- se a implementação for ruim não vai ter alternativa mas pelo menos vai ser compatível. Só o tempo pra dizer ¯\\_(ツ)_/¯.

**Comment by brunocechet** [comment]
Os navegadores da Microsoft nunca trouxeram muita inovação neste mercado. Muitos dos usuários acabam utilizando o IE/Edge por já vir no Windows e mesmo assim ainda perderam muito dos adeptos.

Acho um passo importante para deixar de investir esforços na reinvenção da roda e focar no apoio a comunidade open source, onde todos saem ganhando.
Desde que Satya Nadella assumiu como CEO da Microsoft, a empresa vem se aproximando desde mundo.

Hoje temos o bash do Linux no Windows, abertura de patentes, investimento na Linux Foundation, compra do Github, etc

Vejo uma tendência na Microsoft neste sentido e a adoção do Chromium considero um passo importante e nos beneficia em performance e adoção de padrões.

**Comment by adnux** [comment]
Vai ajudar muito a focar os esforços em solução e não em compatibilidade.
Sem contar que no último chrome dev summit deu pra ver que os caras estão trabalhando em funcionalidades mais 'high level' que outros navegadores nem se propõe a implementar.

**Comment by tcelestino** [comment]
@fabiorocha, vocês desistiram do ChakraCore?

Por um lado acho a incentiva legal para o mundo do desenvolvimento, mas não sei até que ponto teriamos inovações em relação as tecnologias dos browsers.

**Comment by luizbills** [comment]
@tcelestino ChakraCore é somente a Javascript Engine do Edge. Assim como o V8 é do chrome. Provalmente ainda vão usar o Chakra ao invés do v8.

**Comment by iagolaguna** [comment]
@luizbills  mas se:
> Resumindo, o Microsoft Edge vai ser mais um navegador rodando emcima da stack do Chromium, ou parte dela, como o Chrome, Opera, Vivaldi, etc. - @fabiorocha

Não seria rodar em cima da V8?
E sobre a tentativa de fazer o Node.js ser independente da JS Engine essa iniciativa vai morrer também @fabiorocha ?

**Comment by vhfmag** [comment]
Na prática, muda muito pouco:
- É menos um browser pra testar compatiblidade, mas acredito que poucos projetos tem acessos pelo Edge o suficiente pra ele ser uma prioridade na hora de testar (IE parece continuar sendo mais relevante rs)
- É menos uma implementação independente dos padrões da Web, mas eles tão migrando pra uma engine open source, o que permite diversidade a nível de governança, ao invés de a nível de implementação

Dito isso, é um passo simbólico significativo na consolidação do monopólio da web que a Google detem hoje, como foi com a Microsoft anos atrás, e vários sites estão sendo feitos para funcionar *no Chrome*, não na web. A comparação é claramente injusta, porque a Microsoft da época fez a web estagnar (não implementando novos padrões da web e favorecendo tecnologias proprietárias) e usava sua dominância em diversos mercados (browser, hardware, OS) pra atacar concorrentes (nesse caso, o Netscape) agressivamente.

Ainda assim, a web passa pra mão de mais um ditador, ainda que um [ditador benevolente](https://pt.wikipedia.org/wiki/Ditador_benevolente). Por enquanto, funciona maravilhosamente bem e eu amo os padrões que a Google está trazendo pra web (web packaging, portals, feature policy e tantos outros) e o quanto ela investe nela. Mas interesses corporativos mudam, e isso pode não se manter assim pra sempre (spoiler: nunca se mantém). Uma web monopolizada pelo Google é uma web sem privacidade, em que a empresa que decide se um site é relevante (busca do Google) é a mesma que faz o browser em que se navega na web, a mesma que serve de infraestrutura pra esses sites, a mesma que fornece a maior plataforma de videos da web, a mesma que fornece o OS da maioria dos nossos celulares e é responsável pela curadoria dos seus aplicativos, etc. O que dá a ela poder de vigilância e censura imprecendentes na história, o que eu jamais toleraria se viesse de um governo e, por isso, também não tolero vindo de uma empresa.

Enfim, o problema não é o Google (apesar dele ter **diversos** problemas). Qualquer empresa ou organização, na posição de detentora do *monopólio da Web :tm:* tem poder demais e em outros mercados já teria levado processos antitruste. Não ajuda o fato de que a única outra implementação independente de renderização na Web (Gecko, Mozilla) está morrendo e morrendo rápido.

Minha posição é: a mudança é racional e até previsível e não vai mudar muito o status quo, mas é a consolidação de uma situação precária. E é por isso que, me chame de maluco se quiser, mas eu sigo no Firefox, no DuckDuckGo, no NextCloud e saindo de cada vez mais produtos do Google.

---

Textos relevantes:
- https://ferdychristant.com/the-state-of-web-browsers-f5a83a41c1cb
- https://andregarzia.com/2018/12/while-we-blink-we-loose-the-web.html
- https://blog.mozilla.org/blog/2018/12/06/goodbye-edge/
- https://timkadlec.com/remembers/2018-12-04-risking-a-homogenous-web/
- http://www.zeldman.com/2018/12/07/browser-diversity-starts-with-us/

**Comment by fabiorocha** [comment]
Foi mal a demora pessoal (fuso-horário rs), mas quanta resposta boa! Vamos lá:

> Os navegadores da Microsoft nunca trouxeram muita inovação neste mercado. Muitos dos usuários acabam utilizando o IE/Edge por já vir no Windows e mesmo assim ainda perderam muito dos adeptos.
> 
> Acho um passo importante para deixar de investir esforços na reinvenção da roda e focar no apoio a comunidade open source, onde todos saem ganhando.
> Desde que Satya Nadella assumiu como CEO da Microsoft, a empresa vem se aproximando desde mundo. - @brunocechet

Infelizmente tenho que **concordar** com a primeira parte, cara. E sem querer dar desculpas, mas uma das razões disso ter acontecido é que por muitos anos o Windows foi o centro das atenções aqui dentro, e os navegadores daqui tavam muito integrados com o OS e tinham que shippar na mesma cadência do OS. Shippar um navegador a cada 6 meses é devagar demais. Mas o time de web aqui é fenomenal, eu até fico surpreso com o tanto que conseguimos fazer MESMO shippando a cada 6 meses e tendo que seguir o mesmo processo do OS. Vamos ver agora quando começarmos a shippar numa velocidade maior, um software OSS, com uma comunidade ao redor, etc. Tem tudo pra ser mais um projeto igual ao VS Code / TypeScript / etc. Mas você tá certo no segundo ponto, isso tudo começou a mudar quando o Satya assumiu, o cara manda bem.

> @fabiorocha,  vocês desistiram do ChakraCore? - @tcelestino

> @tcelestino ChakraCore é somente a Javascript Engine do Edge. Assim como o V8 é do chrome. Provalmente ainda vão usar o Chakra ao invés do v8. - @luizbills

> E sobre a tentativa de fazer o Node.js ser independente da JS Engine essa iniciativa vai morrer também @fabiorocha ? - @iagolaguna

Vamos usar o V8, mas o desenvolvimento do ChakraCore continua pra outros projetos que não sejam o Edge. @iagolaguna ótima pergunta e eu não sei a resposta oficial, mas vou tentar achar alguém aqui e te respondo.

@vhfmag Muito bem centrado seu comentário, e os links são animais! Aquele primeiro ali é brutalmente honesto e é uma referência muito boa pra se atualizar sobre o estado atual desse mercado.

> Minha posição é: a mudança é racional e até previsível e não vai mudar muito o status quo, mas é a consolidação de uma situação precária.

Vamos ter que esperar pra ver o que muda daqui a alguns anos, mas a minha esperança é que outra empresa de porte competindo de frente emcima da mesma plataforma seja uma coisa boa pra Web em geral.

**Comment by klarkc** [comment]
Vale lembrar que o Chromium é opensource, nada impede de ocorrer um fork no primeiro sinal de "autoritarismo" do Google (o que acho difícil acontecer, dado o histórico da empresa). Um monopólio de uma tecnologia opensource não é um monopólio _evil_, geralmente ocorre justamente pelas vantagens de se ter uma plataforma padrão (como acontece com o Linux, por exemplo).

**Comment by vhfmag** [comment]
@klarkc sim! Vale lembrar que essa é a origem do Blink, afinal de contas: um fork por desentendimentos entre Google e Apple no Webkit. Curioso pra saber se um outro "grande player" usando extensivamente o Chromium (ou V8 e Blink) em seus produtos pode acabar amenizando a dominância da Google, já que teremos mais uma empresa com muitos funcionários dedicados a implementar features e consertar bugs nesses projetos, então interesses de mais de uma organização seriam levados em consideração no desenvolvimento da engine.

Ainda acho que a Web ganharia mais se a Microsoft tivesse decidido trocar o EdgeHTML pelo Gecko, mas sonhar tem seus limites rs.

**Comment by barbier** [comment]
Eu concordo com t[udo o que o Garzia disse neste texto](http://andregarzia.com/2018/12/while-we-blink-we-loose-the-web.html)

**Comment by Rodrigo54** [comment]
Creio que os planos da microsoft vão mais além nessa iniciativa, Tentativas como a Plataforma Universal do Windows (UWP) não se popularizaram e talvez a microsoft invista mais em desenvolvimento com tecnologias da web como as PWA e a plataforma Electron, talvez eles criem um sistema webview como no android mas pra que tudo isso seja possível melhor mesmo seria convergir e não divergir da concorrência

**Comment by evertonthepaula** [comment]
A Microsoft sempre foi pioneira em implementar coisas que depois foram melhor produzidas pelo mercado e evoluíram. Se Edge e IE tem motores que realmente não sustentam a inovação que a MS precise e talvez o custo para chegar lá não valha a pena, acredito que desde que a MS continue explorando novas funcionalidades e empurrando o mercado para frente assim como Google e time da Mozilla vem fazendo a anos, usar Chromium ou não é só um detalhe, até porque a partir de agora com certeza essa ferramenta vai melhorar muito e o ritmo de implementação e suporte pode ser acelerado.

Não me preocupo com isto, pois a MS já provou que sabe entrar e sair de projetos quando necessário, e realmente acho que só temos a ganhar com um Edge mais "parrudo".

**Comment by felquis** [comment]
Acho uma pena que em 2020 ao comprar o Windows 10 não veio com o Edge por padrão, mas sim o IE11.

MAKE EDGE DEFAULT NOW!

**Comment by fabiorocha** [comment]
@felquis como assim cara? 😂 O Edge é pra ser o padrão no Win10, e nas versões mais novas (Win10 2004) é pra ser o Edge baseado no Chromium.

Manda uma mensagem aí em privado (email no perfil), se tiver com problemas! Abs!

**Comment by felquis** [comment]
Iniciei no Windows em Fev/2020 e tinha vindo o IE11 por padrão mesmo.
